### DISCUSSION TITLE
Doubts on Appending Components

### DISCUSSION THREAD

**Discussion by RakulAgn** [GrapesJS/grapesjs#5799] [General]
```
if (component.is('column')) {
        const parentComponent = component.parent();
        if (!parentComponent || !parentComponent.is('column')) return;

        const childComponent = firstChild.clone();
        if (!childComponent)
          throw new Error('Failed to clone the child component.');

        component.remove();
        parentComponent.append(childComponent)
        editor.select(childComponent);
      }
```

The Issue Which i am facing that when i am dragging the component on top of another components its keeping adding at the bottom of the component i know that because of appending its happening is there any alternative solution is there ?

**Comment by artf** [comment]
You can specify the index in [append](https://grapesjs.com/docs/api/component.html#append)

**Reply by RakulAgn** [reply]
then its append on that index only right ? i need to append on any places like default behaviour
